### ISSUE TITLE
BUG:  Failed to upgrade the cluster by sealos run command

### ISSUE THREAD

**Issue by MuJianning** [labring/sealos#3869]
### Sealos Version

v4.3.3

### How to reproduce the bug?

3 virtual machine (virbox, 4G memory, 2 core cpu, 50G storage)

master01 192.168.137.11
master02 192.168.137.12
master03 192.168.137.13

on master01

```shell
echo "deb [trusted=yes] https://apt.fury.io/labring/ /" | tee /etc/apt/sources.list.d/labring.list
apt update
apt install -y sealos=4.3.3

cd ~/.ssh
ssh-keygen -t rsa

ssh-copy-id 192.168.137.12
ssh-copy-id 192.168.137.13

cd ~

sealos run \
  labring/kubernetes:v1.27.4 \
  labring/helm:v3.12.3 \
  labring/calico:v3.25.1 \
  --masters 192.168.137.11,192.168.137.12,192.168.137.13
```

upgrade the cluster by sealos run command

```shell
sealos run labring/kubernetes:v1.27.5
```

failed

```shell
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
W0908 10:53:27.877693   70052 utils.go:69] The recommended value for "healthzBindAddress" in "KubeletConfiguration" is: 127.0.0.1; the provided value is: 0.0.0.0
[preflight] Running pre-flight checks.
[upgrade] Running cluster health checks
[upgrade/health] FATAL: [preflight] Some fatal errors occurred:
        [ERROR ControlPlaneNodesReady]: there are NotReady control-planes in the cluster: [master01]
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher
node/master01 cordoned
node/master01 uncordoned
```

### What is the expected behavior?

The cluster is upgraded.

### What do you see instead?

```shell
[upgrade/health] FATAL: [preflight] Some fatal errors occurred:
        [ERROR ControlPlaneNodesReady]: there are NotReady control-planes in the cluster: [master01]
```

### Operating environment

```markdown
- Sealos version:v4.3.3
- Docker version:Containerd:v1.6.23
- Kubernetes version:v1.27.4, v1.27.5
- Operating system:Ubuntu:v22.04
- Runtime environment:virtual machine (virbox, 4G memory, 2 core cpu, 50G storage)
- Cluster size:3 master, 0 node
- Additional information:helm:v3.12.3, calico:v3.25.1
```

**Comment by cuisongliu** [comment]
[ERROR ControlPlaneNodesReady]: there are NotReady control-planes in the cluster: [master01]

**Comment by cuisongliu** [comment]
node is not ready 。。。

**Comment by MuJianning** [comment]
Besides sealos4.3.3, what's more important is **sealos4.3.0**!

The first installation date of sealos was **August 22nd**.

At that time, the latest version was **4.3.0**.

But sealos4.3.0 does not support **Kubernetes1.27.5**.

Critical operation

```shell
apt install -y sealos=4.3.0

sealos run \
  labring/kubernetes:v1.27.4 \
  labring/helm:v3.12.3 \
  labring/calico:v3.25.1 \
  --masters 192.168.137.11,192.168.137.12,192.168.137.13
```

upgrade the cluster by sealos run command

```shell
sealos run labring/kubernetes:v1.27.5
```

failed

```shell
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
W0908 16:20:49.097467  771353 cluster.go:94] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeadm.k8s.io", Version:"v1beta3", Kind:"ClusterConfiguration"}: strict decoding error: unknown field "APIServer", unknown field "CIImageRepository", unknown field "CIKubernetesVersion", unknown field "CertificatesDir", unknown field "ClusterName", unknown field "ComponentConfigs", unknown field "ControlPlaneEndpoint", unknown field "ControllerManager", unknown field "DNS", unknown field "Etcd", unknown field "FeatureGates", unknown field "ImageRepository", unknown field "KubernetesVersion", unknown field "Networking", unknown field "Scheduler"
W0908 16:20:49.128555  771353 utils.go:69] The recommended value for "authentication.x509.clientCAFile" in "KubeletConfiguration" is: ca.crt; the provided value is: /etc/kubernetes/pki/ca.crt
W0908 16:20:49.128585  771353 utils.go:69] The recommended value for "healthzBindAddress" in "KubeletConfiguration" is: 127.0.0.1; the provided value is: 0.0.0.0
[upgrade/config] FATAL: invalid version ""
To see the stack trace of this error execute with --v=5 or higher
node/master01 cordoned
node/master01 uncordoned
```

**Comment by MuJianning** [comment]
See also:

BUG: failed to upgrade kubernetes version #3585

BUG: sealos upgrade cluster #3798

fix: upgrade cluster #3588

But  the cluster created with sealos4.3.0 cannot be upgraded width sealos4.3.1+.

The ConfigMap kubeadm-config in the kubeadm-config namespace was broken when upgrading the cluster width sealos4.3.0.

How can I repair the broken kubeadm-config?

**Comment by bxy4543** [comment]
known issues need to be added to the document: the upgrade operation requires sealos version 4.3.1 or above. Upgrade is a dangerous operation. Upgrade testing needs to be done before upgrading, and data backup of the cluster needs to be performed.
